modules/blocks/chain.js

```diff
@@ -519,7 +519,7 @@ __private.popLastBlock = function (oldLastBlock, cb) {
 						// WARNING: DB_WRITE
 						modules.transactions.undoUnconfirmed(transaction, cb);
 					}, function (cb) {
-						return setImmediate(cb);
+						modules.transactions.queueTransaction(transaction, cb);
```

Comment:
Upon second review, if we are to use the same "entrance" strategy, please see the following flow:

https://github.com/LiskHQ/lisk/blob/ba4ec3fd69dfc67516cb73370c87f3d68336f2d9/logic/transactionPool.js#L394-L400

https://github.com/LiskHQ/lisk/blob/ba4ec3fd69dfc67516cb73370c87f3d68336f2d9/modules/transactions.js#L401-L403

https://github.com/LiskHQ/lisk/blob/ba4ec3fd69dfc67516cb73370c87f3d68336f2d9/logic/transactionPool.js#L470-L492

Note that we process and verify the transaction **before** queuing it. Therefore here I think we should instead use `modules.transactions.processUnconfirmedTransaction`, which is already exposed publicly.

We also need to remember the transaction we are "re-queuing" may be at the same time received as a broadcast.

Reply:
Done, changed to `TransactionPool.prototype.receiveTransactions` to take advantage of transaction array